airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -381,175 +381,302 @@ static Optional<ContainerOrchestratorConfig> getContainerOrchestratorConfig(fina
     }
   }
 
-  private static void launchWorkerApp(final Configs configs, final DSLContext configsDslContext, final DSLContext jobsDslContext) throws IOException {
-    MetricClientFactory.initialize(MetricEmittingApps.WORKER);
+  private static AirbyteApiClient getApiClient(final Configs configs) {
+    final var authHeader = configs.getAirbyteApiAuthHeaderName();
+
+    // control plane workers communicate with the Airbyte API within their internal network, so https
+    // isn't needed
+    final var scheme = configs.isControlPlaneWorker() ? "http" : "https";
+
+    LOGGER.debug("Creating Airbyte Config Api Client with Scheme: {}, Host: {}, Port: {}, Auth-Header: {}",
+        scheme, configs.getAirbyteApiHost(), configs.getAirbyteApiPort(), authHeader);
+
+    final AirbyteApiClient airbyteApiClient = new AirbyteApiClient(
+        new io.airbyte.api.client.invoker.generated.ApiClient()
+            .setScheme(scheme)
+            .setHost(configs.getAirbyteApiHost())
+            .setPort(configs.getAirbyteApiPort())
+            .setBasePath("/api")
+            .setRequestInterceptor(builder -> {
+              builder.setHeader("User-Agent", "WorkerApp");
+              if (!authHeader.isBlank()) {
+                builder.setHeader(authHeader, generateAuthToken());
+              }
+            }));
+    return airbyteApiClient;
+  }
+
+  /**
+   * Generate an auth token based on configs. This is called by the Api Client's requestInterceptor
+   * for each request.
+   *
+   * For Data Plane workers, generate a signed JWT as described here:
+   * https://cloud.google.com/endpoints/docs/openapi/service-account-authentication
+   *
+   * Otherwise, use the AIRBYTE_API_AUTH_HEADER_VALUE from EnvConfigs.
+   */
+  private static String generateAuthToken() {
+    if (configs.isControlPlaneWorker()) {
+      // control plane workers communicate with the Airbyte API within their internal network, so a signed
+      // JWT isn't needed
+      return configs.getAirbyteApiAuthHeaderValue();
+    } else if (configs.isDataPlaneWorker()) {
+      try {
+        final Date now = new Date();
+        final Date expTime = new Date(System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(JWT_TTL_MINUTES));
+        final String saEmail = configs.getDataPlaneServiceAccountEmail();
```

Comment:
I think this means that the authentication is globally shared for all workers in the data plane - meaning any worker can modify the state for any workspace and any connection... that seems too open?  Perhaps the thing to do is mux the the service account email with the workspace to scope the grant down - `foo@googleservices.com` beccomes `foo+workspace_id@googleservices.com`

Reply:
@git-phu and @davinchia, did we ever brainstorm a way to scope our service accounts down further? I think workspace_id probably won't work because once we launch multi-cloud functionality to all of our customers, any worker could theoretically need to handle tasks for any workspace. 

If a customer creates a new workspace and requests that their sync runs in the EU, for example, we want existing workers to be able to handle that work without changing any configurations or re-deploying

Reply:
@evantahler when we start thinking about Hybrid deploy, I think we'll absolutely want to scope down a customer's on-prem Data Plane workers so that they can only access their own workspace. But for the multi-cloud MVP that enables Airbyte Cloud in the EU, Airbyte owns and and controls every Data Plane worker

Reply:
@pmossman think we can handle this as part of the routing spec. Once we make the relevant database changes so we can associate a connection to a task queue, we can use that knowledge to apply permission on the specific dataplane email the same way we permission any users now. Until we have that association, enforcing this is tough.

@evantahler good point, though think we can add this later as we polish things up.

Reply:
Continued in child PR here: https://github.com/airbytehq/airbyte/pull/15904/files#r953192438